src/centralstore/incremental/NodeManager.cpp

```diff
@@ -0,0 +1,254 @@
+        st_ino     file serial number
+        st_mode    mode of file (see below)
+        st_nlink   number of links to the file
+        st_uid     user ID of file
+        st_gid     group ID of file
+        st_rdev    device ID (if file is character or block special)
+        st_size    file size in bytes (if file is a regular file)
+        st_atime   time of last access
+        st_mtime   time of last data modification
+        st_ctime   time of last status change
+        st_blksize a filesystem-specific preferred I/O block size for
+                        this object.  In some filesystem types, this may
+                        vary from file to file
+        st_blocks  number of blocks allocated for this object
+    */
+    struct stat result;
+    if (stat(path.c_str(), &result) == 0) {
+        node_manager_logger.debug("Size of the " + path + " is " + std::to_string(result.st_size));
+    } else {
+        node_manager_logger.error("Error while reading file stats in " + path);
+        return -1;
+    }
+
+    return result.st_size;
+}
+
+/**
+ * @Deprecated use NodeBlock.get() instead
+ **/
+NodeBlock *NodeManager::get(std::string nodeId) {
+    NodeBlock *nodeBlockPointer = NULL;
+    int debug = std::stoi(nodeId);
+    if (this->nodeIndex.find(nodeId) == this->nodeIndex.end()) {  // Not found
+        return nodeBlockPointer;
+    }
+    unsigned int nodeIndex = this->nodeIndex[nodeId];
+    const unsigned int blockAddress = nodeIndex * NodeBlock::BLOCK_SIZE;
+    NodeBlock::nodesDB->seekg(blockAddress);
+    unsigned int edgeRef;
+    unsigned int propRef;
+    char usageBlock;
+    char label[NodeBlock::LABEL_SIZE];
+
+    if (!NodeBlock::nodesDB->get(usageBlock)) {
+        node_manager_logger.error("Error while reading usage data from block " + std::to_string(blockAddress));
+    }
+
+    if (!NodeBlock::nodesDB->read(reinterpret_cast<char *>(&edgeRef), sizeof(unsigned int))) {
+        node_manager_logger.error("Error while reading edge reference data from block " + std::to_string(blockAddress));
+    }
+
+    if (!NodeBlock::nodesDB->read(reinterpret_cast<char *>(&propRef), sizeof(unsigned int))) {
+        node_manager_logger.error("Error while reading prop reference data from block " + std::to_string(blockAddress));
+    }
+
+    if (!NodeBlock::nodesDB->read(&label[0], NodeBlock::LABEL_SIZE)) {
+        node_manager_logger.error("Error while reading label data from block " + std::to_string(blockAddress));
+    }
+    bool usage = usageBlock == '\1';
+    node_manager_logger.debug("Label = " + std::string(label));
+    node_manager_logger.debug("Length of label = " + std::to_string(strlen(label)));
+    node_manager_logger.debug("DEBUG: raw edgeRef from DB (disk) " + std::to_string(edgeRef));
+
+    nodeBlockPointer = new NodeBlock(nodeId, blockAddress, propRef, edgeRef, label, usage);
+
+    node_manager_logger.debug("DEBUG: nodeBlockPointer after creating the object edgeRef" +
```

Comment:
Here `edgeRef"` has to be changed to `edgeRef "`

Reply:
Fixed in https://github.com/miyurud/jasminegraph/pull/115/commits/ed31ba0af33e9629173808960eef3cfe90c5dbe9

Reply:
Done